tools/lsp/properties.rs

```diff
@@ -10,7 +10,9 @@ use std::collections::HashSet;
 #[cfg(target_arch = "wasm32")]
 use crate::wasm_prelude::*;
 
-#[derive(serde::Deserialize, serde::Serialize, Debug, PartialEq)]
+pub type OffsetToPositionMapper<'a> = Box<dyn FnMut(u32) -> lsp_types::Position + 'a>;
```

Comment:
It is possible to avoid the Box if you use a trait. A box with a lifetime is IMO often (not always) an oxymoron, since the box doesn't truly own. You can for example use a trait and define it for a tuple of the cache and the uri: https://gist.github.com/bebef6bc284a4e7ce6e0a0911764ce75